### PR TITLE
Use java toolchain in generated convention plugin

### PR DESCRIPTION
**Pull request by grossws**
Part-of #14599

### Context
`gradle init` adds java toolchain in the generated `build.gradle.kts` but not to `<package>.java-common-conventions.gradle.kts` convention plugin (generated when `--split-project` is used).

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation

### CODE REVIEW

File: subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/MultiProjectJvmApplicationInitIntegrationTest.groovy [hunk 1]
```diff
@@ -52,13 +54,20 @@ abstract class AbstractMultiProjectJvmApplicationInitIntegrationTest extends Abs
...
+        buildLogicFolder.assertHasDescendants(
             settingsFile,
             buildFile,
-            "src/main/${dsl.id}/some.thing.${dsl.fileNameFor("${language}-common-conventions")}",
+            commonConventionsPath,
             "src/main/${dsl.id}/some.thing.${dsl.fileNameFor("${language}-application-conventions")}",
             "src/main/${dsl.id}/some.thing.${dsl.fileNameFor("${language}-library-conventions")}",
         )
+        buildLogicFolder.file(commonConventionsPath).assertContents(
+            containsText("JavaLanguageVersion.of(" + Jvm.current().javaVersion.majorVersion + ")")
```
Comment (jvandort) [inline subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/MultiProjectJvmApplicationInitIntegrationTest.groovy:69]: Can we perform a stronger assertion here? IMO we should be testing for the existence of the entire `toolchain` block and its contents as opposed to testing that `JavaLanguageVersion` exists _somewhere_ in the file without the additional context

**Review by jvandort — APPROVED** [review]
Would like to see stricter testing. LGTM otherwise

File: subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/MultiProjectJvmApplicationInitIntegrationTest.groovy (hunk 1 above)
Comment (jvandort) [reply subprojects/build-init/src/integTest/groovy/org/gradle/buildinit/plugins/MultiProjectJvmApplicationInitIntegrationTest.groovy:69]: Ahh I see. I'm fine the way it is then

**Review by octylFractal — APPROVED** [review]

**Comment by octylFractal** [comment]
@bot-gradle test and merge

**Pull request merged**